src/odemis/driver/test/xt_client_test.py

```diff
@@ -616,6 +616,21 @@ def test_secondary_stigmator(self):
         # Set back to the initial stigmator value, so the system is not misaligned after finishing the test.
         self.microscope.set_secondary_stigmator(*init_stig)
 
+    def test_pattern_stigmator(self):
+        """Test getting and setting the secondary stigmator."""
+        if self.xt_type != 'xttoolkit':
+            self.skipTest("This test needs XTToolkit to run.")
+        stig_range = self.microscope.pattern_stigmator_info()["range"]
+        init_stig = self.microscope.get_pattern_stigmator()
+        stig_x, stig_y = (init_stig[0] + 1e-3, init_stig[1] + 1e-3)
+        stig_x = stig_x if stig_range['x'][0] < stig_x < stig_range['x'][1] else init_stig[0] - 1e-3
+        stig_y = stig_y if stig_range['y'][0] < stig_y < stig_range['y'][1] else init_stig[1] - 1e-3
+        self.microscope.set_pattern_stigmator(stig_x, stig_y)
+        self.assertEqual(self.microscope.get_pattern_stigmator()[0], stig_x)
+        self.assertEqual(self.microscope.get_pattern_stigmator()[1], stig_y)
+        # Set back to the initial stigmator value, so the system is not misaligned after finishing the test.
+        self.microscope.set_pattern_stigmator(*init_stig)
+
```

Comment:
Why didn't/don't we add something like this in any of the tests?

with self.assertRaises(OSError):
               self.microscope.set_pattern_stigmator(10e10, 10e10)
# Check if values didn't change:
self.assertEqual(self.microscope.get_pattern_stigmator()[0], stig_x)
self.assertEqual(self.microscope.get_pattern_stigmator()[1], stig_y)

Reply:
That's a good idea. There are a few tests that check if an exception is raised (for instance `test_set_scan_field_size`). I think it was not added in other tests because we simply forgot.

Reply:
I will add it for the client and adapter.
@Kleijwegt Is there a reason to specifically check for OSError, or is an Exception enough. We have in some places such a check in the client, but then check generically for an Exception.

Reply:
On the adapter side we're raising an IOError, which Pyro5 will turn into an OSError. So we're expecting an OSError. It's cleaner to check for the specific error instead of a generic error. If something else breaks on the adapter side and for instance a ValueError is raised, while we check for a generic error, the test case will pass. If we a ValueError is raised and we check for a OSError, this testcase will fail.

The other test cases should probably also check for OSErrors, and not Exceptions

Reply:
Ok then let's change it.